src/rust/engine/fs/src/lib.rs

```diff
@@ -764,36 +976,85 @@ pub trait VFS<E: Send + Sync + 'static>: Clone + Send + Sync + 'static {
     &self,
     path_glob: PathGlob,
     exclude: &Arc<Gitignore>,
-  ) -> BoxFuture<(Vec<PathStat>, Vec<PathGlob>), E> {
-    match path_glob {
+  ) -> BoxFuture<SingleExpansionResult, E> {
+    // If we were to match on &path_glob (which I just tried), it seems like it's impossible to move
+    // something while borrowing it, even if the move happens at the end of the borrowed scope. That
+    // seems like it doesn't need to happen? I think there is no way to avoid the clone here,
+    // however, because directory_listing() returns a future and therefore can use only static
```

Comment:
Rather than eagerly cloning the entire `path_glob` here, you should clone only the fields that need to move into that closure: that is, after destructuring the path_glob, if you need two copies of any particular field, you should be able to clone them directly (and chances are it will be less than "the entire glob").

Reply:
Yes -- I cloned it here (in `expand_single()`) because it means we can then just move the `sourced_globs` argument, and because we are definitely moving all the elements of either variant of a `PathGlob`. I've noted your comments for future reference!